ui/src/table.tsx

```diff
@@ -335,6 +340,7 @@ const
   DataTable = React.forwardRef(({ model: m, onFilterChange, items, filteredItems, selection, selectedFilters, isMultiple, isSingle, groups, expandedRefs, onSortChange, setFiltersInBulk }: DataTable, ref) => {
     const
       [colContextMenuList, setColContextMenuList] = React.useState<Fluent.IContextualMenuProps | null>(null),
+      [sortCols, setSortCols] = React.useState<Dict<S> | null>(null),
```

Comment:
I tried using `col.iconName` initially, but it didn't behave as I expected.  All elements of the column header are now handled in `onRenderHeader`, which includes the sorting icon - this is needed to control the placement of each of the elements of the column header. But when `col.iconName` is set, then Fluent tries to render the sort icon, even though it is already defined in `onRenderHeader`. I tried various things to work around this, but no luck. That's why I decided to define a new state variable. Let me know if this makes sense. Thanks!

Reply:
@mturoci, Since I know your busy, I decided to dig a little deeper into your question about why I was unable to use `col.iconName` to hold the state of the sort icon sorting direction. I created a small Fluent app outside of Wave to isolate the situation. It is a new table component that wraps the Fluent `DetailsList` component. The new component is very simple and dumb, with just 3 columns and 5 rows of data. Each column has a ContextMenu that behaves like it would for filtering, but it doesn't actually filter anything. I also keep track of the number of selections for each column and print those at the top.

Here's a screen show of what this looks like:

![issue1689_iconName](https://github.com/user-attachments/assets/3e58b120-4d57-40f3-9f1b-e0fd40a73bca)

Note: You'll find a code snipit where these 3 columns are defined at the bottom of this comment.

The first column is the plainest with no custom renderer used. It has a Sort icon, defined using the `col.iconName` property, the column name, an onColumnClick handler, and a dropdown. The Sort icon doesn't do anything in this example.

The second column uses `onRenderHeader` to display the column header. In onRenderHeader, the column name, Sort icon, filter icon, and selection count are displayed. As with the 1st column, the Sort icon doesn't do anything, just there to display.

The third column is a duplicate of the 2nd column, with the additon of setting `col.iconName`. You'll notice the only thing visible in the 3rd column is the Sort icon. The behavior is the same whether I set this before or after setting the `onRenderHeader` property.

When I looked in the Chrome dev tools, I could see that the `<div>` in onRenderHeader is in the document, but it is on the next line below the header and hidden behind the 1st row of data.

I didn't dig into the Fluent code, but it seems to me that Fluent doesn't behave well if you define a column with both an `iconName` and `onRenderHeader`. And since we need to display the sort icon in `onRenderHeader`, then we need to define a new state variable to hold the sorting direction.

I hope this all makes sense and helps. And if I'm overlooking something, please let me know. Thanks!

Here's the snipit:

```typescript
    const columns: IColumn[] = [
        { key: 'column1', name: 'Name', fieldName: 'name', minWidth: 100, 
          iconName: 'Sort',   
          onColumnClick: (ev, column) => {
            onNameColumnContextMenuHandler(column, ev);
          },
          columnActionsMode: ColumnActionsMode.hasDropdown,
          isResizable: true,
        },        
        { key: 'column2', name: 'Value', fieldName: 'value', minWidth: 100, 
          onRenderHeader: (props?: IDetailsColumnProps) => {
            if (!props) return null
            return (
                <div>
                    {props.column.name}
                    <Icon iconName='Sort'/>
                    <Icon iconName='ChevronDown'
                          onClick={(ev: React.MouseEvent<HTMLElement>) =>{
                            onValueColumnContextMenuHandler(props.column, ev);
                          }}
                    />
                    <span>{valueCheckedCount}</span>
                </div>
            ) 
          },
          isResizable: true, 
        },             
        { key: 'column3', name: 'Company', fieldName: 'company', minWidth: 100, 
            iconName: 'Sort', 
            onRenderHeader: (props?: IDetailsColumnProps) => {
              if (!props) return null
              return (
                  <div>
                      {props.column.name}
                      <Icon iconName='Sort'/>
                      <Icon iconName='ChevronDown'
                            onClick={(ev: React.MouseEvent<HTMLElement>) =>{
                              onCompanyColumnContextMenuHandler(props.column, ev);
                            }}
                      />
                      <span>{companyCheckedCount}</span>
                  </div>
              ) 
            },
            isResizable: true,
            // iconName: 'Sort', 
          }, 
    ];
```

Reply:
@mturoci One more thing, it would be possible to store the state of the sorting direction in `col.iconName`, but that would require 2 things, which I think are a bit of a hack:
1. The `<div>` would need to be defined with `display: 'inline-block'` to prevent the div from starting on the next line - or use a `<span>`; 
2. The Sort icon set with `col.iconName` would have to be defined as both hidden and with a width of 0 so that it wouldn't ever display

The result would be something like this:

![issue1689_sortIcon2](https://github.com/user-attachments/assets/c9e2f5f8-af35-405d-ab85-634c9b131c49)

While this might technically work, it seems more intuitive to me to just define a new state variable for the sorting direction. 

Anyway, wanted to share this. Let me know what you think. thanks!

Reply:
> seems to me that Fluent doesn't behave well if you define a column with both an iconName and onRenderHeader.

That's enough of a reason to me. Didn't check myself, but as long as the new code doesn't cause any regressions, it's fine as is. Thanks for all the digging!